federation/README.md

```diff
@@ -0,0 +1,318 @@
+
+# Overview
+
+This tutorial shows how to authenticate two SPIFFE-identified workloads that are identified by two different SPIRE Servers.
+
+In this tutorial you will learn how to:
+
+* Configure each SPIRE Server to expose it's SPIFFE Federation bundle endpoint using SPIFFE authentication and WebPKI authentication.
+* Configure the SPIRE Servers to retrieve trust bundles from each other (using the federate_with configuration section).
+* Bootstrap federation between two SPIRE Servers using different trust domains.
+* Create registration entries for the workloads so that they can federate with other trust domain.
+
+# Prerequisites
+
+* Two SPIRE Servers instances.
+* Two SPIRE Agents, one connected to one SPIRE Server, and the second connected to the other SPIRE Server.
+* Two workloads that needs to communicate each other via mTLS, and use the Workload API to get SVIDs and trust bundles.
+
+# Scenario
+
+Let's say we have a stock broker's webapp that wants to display stock quotes fetched from a stocks market webservice provider. The scenario goes as follows:  
+
+1. The user enters the broker's webapp stock quotes URL in a browser.
+2. The webapp workload receives the request and makes an HTTP request for quotes to the stocks market service using mTLS.
+3. The stocks market service receives the request and sends the quotes in the response.
+4. The webapp renders the stock quotes page using the returned quotes and sends it to the browser.
+5. The browser displays the quotes to the user. The webapp includes some javascript to refresh the page every 1 second, so every second these steps are executed again.
+
+In addition to the above and for the rest of this tutorial, we are going to assume the following trust domain names for their SPIRE installations: `broker.org` and `stocksmarket.org`.  
+Also, the applications access the WorkloadAPI directly to get SVIDs and trust bundles, meaning there are no proxies in the scenario described.
+
+# Configuring SPIFFE Federation endpoint
```

Comment:
The issue is that it is much easier to see federation in action using SPIFFE auth in both sides, because we can do it by just using docker-compose, without requiring to have ownership over any domain name.

Reply:
Ah, so the docker-compose demo only does the SPIFFE auth, but you are also explaining the alternate method using Web PKI. This makes sense, so let's keep the existing structure. I think we should just explicitly explain that in the doc. So in the first Web PKI section, insert a new first sentence in the first paragraph. Something like:

Although not shown in the Docker Compose tutorial, you could make the changes shown here to use Web PKI for authentication instead of SPIFFE auth.

Reply:
What if I add something like that under the `Federation Example Using SPIFFE Authentication with SPIRE 0.11.0` H1?

Reply:
Sure, you can put it in the explanation of the docker-compose demo as you suggest.